app/src/ui/lib/theme-change-monitor.ts

```diff
@@ -0,0 +1,60 @@
+import { remote } from 'electron'
+import { ApplicationTheme } from './application-theme'
+import { IDisposable, Disposable, Emitter } from 'event-kit'
+import { supportsDarkMode, isDarkModeEnabled } from './dark-theme'
+
+class ThemeChangeMonitor implements IDisposable {
+  private readonly emitter = new Emitter()
+
+  private subscriptionID: number | null = null
+
+  public constructor() {
+    this.subscribe()
+  }
+
+  public dispose() {
+    this.unsubscribe()
+  }
+
+  private subscribe = () => {
+    if (!supportsDarkMode()) {
+      return
+    }
+
+    this.subscriptionID = remote.systemPreferences.subscribeNotification(
+      'AppleInterfaceThemeChangedNotification',
+      this.onThemeNotificationFromOS
+    )
+  }
+
+  private onThemeNotificationFromOS = (event: string, userInfo: any) => {
+    const darkModeEnabled = isDarkModeEnabled()
+
+    const theme = darkModeEnabled
+      ? ApplicationTheme.Dark
+      : ApplicationTheme.Light
+    this.emitThemeChanged(theme)
+  }
+
+  private unsubscribe = () => {
+    if (this.subscriptionID !== null) {
+      remote.systemPreferences.unsubscribeNotification(this.subscriptionID)
```

Comment:
`this.subscriptionID` should probably be set to `null` after you unsubscribe from the notification.